src/lldp_syncd/daemon.py

```diff
@@ -109,35 +111,84 @@ class ChassisIdSubtypeMap(int, Enum):
         # chassis = int(LldpChassisIdSubtype.chassisComponent) # (unsupported by lldpd)
         local = int(LldpPortIdSubtype.local)
 
+    def get_sys_capability_list(self, if_attributes):
+        """
+        Get a list of capabilities from interface attributes dictionary.
+        :param if_attributes: interface attributes
+        :return: list of capabilities
+        """
+        try:
+            # [{'enabled': ..., 'type': 'capability1'}, {'enabled': ..., 'type': 'capability2'}]
+            capability_list = if_attributes['chassis'].values()[0]['capability']
+            # {'enabled': ..., 'type': 'capability'}
+            if not isinstance(capability_list, list):
```

Comment:
>capability_list [](start = 30, length = 15)

Is this case ever possible? If not, just return None.
I am worry about the blindness to error will mess parse_sys_capabilities()

Reply:
I think it's possible for a device to have only 1 supported or enabled capability. And in the same way in tests/subproc_outputs/lldpctl_mgmt_only.json we have a dictionary of 1 interface under ['lldp]['interfaces'], where for many interfaces a list would be present. It is treated in a similar fashion in parse_update() L#169.

Reply:
Could you provide proof for your thought? Either source code and observed data. I don't see the case inside test data.

Even it is possible, I suggest test capability_list type positively instead of 'not list'.

---
In reply to: [203343589](https://github.com/Azure/sonic-dbsyncd/pull/5#discussion_r203343589) [](ancestors = 203343589)